### PR TITLE
feat(docs): Add documentation about the accepted query parameters.

### LINKED ISSUE #2179: document all possible query parameters
Right now we have no documentation on the myriad query parameters that are accepted to signin/signup and what they are used for. We should document these.

Ref https://github.com/mozilla/fxa-content-server/issues/1898

### PR DESCRIPTION
**Pull request by shane-tomlinson**
fixes #2179

### CODE REVIEW

**Comment by shane-tomlinson** [comment]
@zaach or @vladikoff - mind doing a quick scan? Documentation only.

**Comment by zaach** [comment]
`redirectTo` would be used for OAuth, but we currently don't use it at all. It would allow RPs to dynamically set an endpoint to redirect to after the OAuth flow, as long as the prefix of the URL matched with whatever `redirect` is stored on the OAuth server.

`allowCachedCredentials` is not a parameter, rather, it's something set by the relier if the `email` parameter is set to `blank`. We can note under the `email` parameter that if it's set to `blank`, a empty sign in form will be displayed– no suggested accounts will appear.

**Comment by shane-tomlinson** [comment]
Thanks for the feedback/info @zaach. Updated.

File: docs/query-params.md [hunk 1]
```diff
@@ -0,0 +1,216 @@
...
+## Email verification parameters
+
+### `code`
+Used in the verification flows to specify the verification code.
+
+#### When to use
+Should not be used by reliers.
+
+### `uid`
+Used in the verification flows to specify the user id of the user being verified.
```
Comment (zaach) [inline docs/query-params.md]: Hm, RPs can also use `uid` when loading a settings page to set which account should be used. If the account data for `uid` is not in localStorage, they're taken to regular signin (not force_auth), but the RP can suggest a prefill email using the `email` param.

**Comment by shane-tomlinson** [comment]
@zaach - updated with your notes about `uid`.

File: docs/query-params.md [hunk 2]
```diff
@@ -0,0 +1,228 @@
...
+Specify an OAuth state token.
+
+#### When to specify
+When authenticating a user with OAuth.
+
+* /oauth/signin
+* /oauth/signup
+* /oauth/force_auth
+
+### `style`
```
Comment (shane-tomlinson) [inline docs/query-params.md]: This is in anticipation of #2236

**Comment by zaach** [comment]
:+1: nice

**Pull request merged**